runtime/autoload/health/provider.vim

```diff
@@ -497,6 +485,74 @@ function! s:check_for_pyenv() abort
   return [pyenv_path, pyenv_root]
 endfunction
 
+" Locate Python executable by running invocation and checking
+" sys.executable.
+function! s:locate_pythonx(invocation) abort
+  return s:normalize_path(system(a:invocation
+    \ . ' -c "import sys; sys.stdout.write(sys.executable)"'))
+endfunction
+
+" If $VIRTUAL_ENV is set, check whether its Python executables will be
+" the first on the $PATH of both Neovim and subshells spawned from
+" Neovim.
+function! s:check_active_virtualenv() abort
+  call health#report_start('Python active virtualenv')
+  let errors = []
+  " hints are kept as Dictionary keys in order to discard duplicates
+  let hints = {}
+  " the virtualenv should contain some Python executables, and those
+  " executables should be first both on Neovim's path and on the path of
+  " subshells launched from Neovim
+  let bin_dir = has('win32') ? '/Scripts' : '/bin'
+  let venv_pythonxs = glob($VIRTUAL_ENV . bin_dir . '/python*', v:true, v:true)
+  if len(venv_pythonxs)
+    for venv_pythonx in venv_pythonxs
+      let venv_pythonx = s:normalize_path(venv_pythonx)
+      let pythonx_basename = fnamemodify(venv_pythonx, ':t')
+      let neovim_pythonx = s:locate_pythonx(exepath(pythonx_basename))
+      let subshell_pythonx = s:locate_pythonx(pythonx_basename)
+      if venv_pythonx !=# neovim_pythonx
+        call add(errors, 'according to Neovim''s $PATH, the ' . pythonx_basename
+          \ . ' executable is found outside the virtualenv, here: ' . neovim_pythonx)
+        let hint = 'Problems with Neovim''s $PATH are caused by the virtualenv not being '
```

Comment:
seems distracting to refer to `$PATH` as "neovim's PATH". It's not really related to Nvim, it is just the typical concept of $PATH that all processes are subject to.

Reply:
I realize it's wordy, I just wanted to drive home that there's a distinction between what Neovim thinks is the PATH and what subshells launched from Neovim think (they inherit it from Neovim, but may modify it via rc files). Which is the source of some of the issues this check is trying to troubleshoot. But maybe it's clear enough even without, and the terser the message, the more likely people will actually read it :)

Reply:
Anyway, I like the way you've reworded the warnings and hints, they're shorter, clearer and actionable :) Resolving this.